src/Microsoft.Azure.WebJobs/StorageAccountAttribute.cs

```diff
@@ -27,14 +27,22 @@ public sealed class StorageAccountAttribute : Attribute
         ///   to any INameResolver registered on the JobHostConfiguration to resolve the actual setting name to use.
         /// - Can be an app setting or connection string name of your choosing.
         /// </param>
-        public StorageAccountAttribute(string account)
+        /// <param name="supported">A flags enum indicating which types of Azure Storage accounts are
+        /// supported by this attribute.</param>
+        public StorageAccountAttribute(string account, StorageAccountTypes supported = StorageAccountTypes.All)
         {
             Account = account;
+            Supported = supported;
         }
 
         /// <summary>
         /// Gets the name of the Azure Storage connection string to use.
         /// </summary>
         public string Account { get; private set; }
+
+        /// <summary>
+        /// Gets the disabled storage account types.
+        /// </summary>
+        public StorageAccountTypes Supported { get; private set; }
```

Comment:
The motivation is to allow functions to disable blobtriggers on premium.  I agree with keeping the public api simple (this probably doesn't have much value to an average joe user).  

Is it preferable to have a way of detecting whether the web jobs sdk is running in a functions environment?  @fabiocav argued against this, hence the attribute.